examples/components/data_table/selectable_usage.html

```diff
@@ -10,14 +10,14 @@
 </md-toolbar>
 <md-data-table layout-fill [selectable]="true" (onSelectableChange)="change($event)">
   <thead>
-  <tr md-data-table-header-row>
+  <tr md-data-table-header-selectable-row>
     <th class="md-text-cell">Material</th>
     <th>Quantity</th>
     <th>Unit price</th>
   </tr>
   </thead>
   <tbody>
-  <tr md-data-table-row *ngFor="#material of materials" [selectable-value]="material.id">
```

Comment:
My first thought was your second proposition too. Then I've watched my code and see the size of the component. ATM, it's only a component that manage selection but tomorrow there will be inline form, etc. so the component will become even bigger.
One solution is to split component/directive by feature instead of markup.